mlir/lib/Dialect/Tosa/IR/TosaOps.cpp

```diff
@@ -610,15 +647,17 @@ LogicalResult tosa::PadOp::inferReturnTypeComponents(
     }
 
     outputShape.resize(paddingShape.getDimSize(0), ShapedType::kDynamic);
-    inferredReturnShapes.push_back(ShapedTypeComponents(outputShape));
+    inferredReturnShapes.push_back(
```

Comment:
I have updated PadOp to use InferTensorType so that the inferred output shape and type is validated against the actual output. For this to work, the referred output shape from PadOp::inferReturnTypeComponents should always have a type or else it throws an error. 
Since input and output types are always same for PadOp, I am using the inputType as element type for the output.